radixdlt-java/src/main/java/com/radixdlt/client/core/ledger/RadixParticleStore.java

```diff
@@ -1,34 +1,35 @@
 package com.radixdlt.client.core.ledger;
 
 import com.radixdlt.client.atommodel.accounts.RadixAddress;
-import com.radixdlt.client.core.atoms.AtomObservation;
 import io.reactivex.Observable;
 
 import java.util.concurrent.ConcurrentHashMap;
 
 public class RadixParticleStore implements ParticleStore {
 	private final AtomStore atomStore;
-	private final ConcurrentHashMap<RadixAddress, Observable<TransitionedParticle>> cache = new ConcurrentHashMap<>();
+	private final ConcurrentHashMap<RadixAddress, Observable<ParticleObservation>> cache = new ConcurrentHashMap<>();
 
 	public RadixParticleStore(AtomStore atomStore) {
 		this.atomStore = atomStore;
 	}
 
 	@Override
-	public Observable<TransitionedParticle> getParticles(RadixAddress address) {
-		// TODO: use https://github.com/JakeWharton/RxReplayingShare to disconnect when unsubscribed
+	public Observable<ParticleObservation> getParticles(RadixAddress address) {
 		return cache.computeIfAbsent(address, addr -> atomStore.getAtoms(address)
-			.filter(AtomObservation::hasAtom)
-			.flatMap(observation -> Observable.<TransitionedParticle>create(emitter -> {
-				try {
-					observation.getAtom().spunParticles()
-						.filter(s -> s.getParticle().getShardables().contains(address))
-						.forEach(s -> emitter.onNext(TransitionedParticle.fromSpunParticle(s, observation.getType())));
-				} catch (Throwable e) {
-					emitter.onError(e);
+			.flatMap(observation -> Observable.create(emitter -> {
+				if (observation.hasAtom()) {
+					observation.getAtom().spunParticles().filter(s -> s.getParticle().getShardables().contains(address))
+						.forEach(s -> {
+							final TransitionedParticle tp =
+								TransitionedParticle.fromSpunParticle(s, observation.getType());
+
+							final ParticleObservation p = ParticleObservation.ofParticle(tp);
```

Comment:
Petty - technically this should be called `ParticleObservation.ofTransitionedParticle`